src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
+const CARNIVORE_BITE = 50;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - Instead of using indexOf and splice to remove a dead animal from the alive array, consider using filter. This will avoid the nesting loops performance issue. You can implement this directly in the die method or within the bite method after checking the health of the animal.